### PR TITLE
Deprecate/remove contribs: atom, jsrouting, limiter, testtools

### PR DESCRIPTION
**Pull request by lepture**
Via https://github.com/pallets/werkzeug/issues/4

Should we deprecate other modules:

- iterio
- lint
- profiler

### CODE REVIEW

File: werkzeug/contrib/limiter.py [hunk 1]
```diff
@@ -31,7 +31,12 @@ class StreamLimitMiddleware(object):
     """
 
     def __init__(self, app, maximum_size=1024 * 1024 * 10):
-        warn(DeprecationWarning('This middleware is deprecated'))
+        warnings.warn(
+            'werkzeug.contrib.limiter is deprecated as of version 0.15 and'
```
Comment (RonnyPfannschmidt) [inline werkzeug/contrib/limiter.py]: based on the doc-string this should be `0.9` instead of `0.15`

File: werkzeug/contrib/jsrouting.py [hunk 2]
```diff
@@ -18,9 +18,18 @@
...
             raise RuntimeError('simplejson required for jsrouting')
 
+import warnings
 from inspect import getmro
 from werkzeug.routing import NumberConverter
 from werkzeug._compat import iteritems
+from werkzeug.contrib import WerkzeugContribDeprecationWarning
+
+warnings.warn(
+    'werkzeug.contrib.jsrouting is deprecated as of version 0.15 and will'
```
Comment (RonnyPfannschmidt) [inline werkzeug/contrib/jsrouting.py]: for this one i wonder if deprecation is the right path forward - are there users and/or replacements

but i'm `-0` so it probably could go

File: werkzeug/contrib/cache.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ class BaseCache(object):
     def __init__(self, default_timeout=300):
         warnings.warn(
             'Werkzeug cache is deprecated as of version 0.15 and will'
-            ' be removed in version 1.0.',
+            ' be removed in version 1.0. Check "cachelib" instead.',
```
Comment (RonnyPfannschmidt) [inline werkzeug/contrib/cache.py]: :+1:

File: werkzeug/contrib/atom.py [hunk 4]
```diff
@@ -21,12 +21,20 @@ def atom_feed(request):
...
 from datetime import datetime
-
 from werkzeug.utils import escape
 from werkzeug.wrappers import BaseResponse
+from werkzeug.contrib import WerkzeugContribDeprecationWarning
 from werkzeug._compat import implements_to_string, string_types
 
+warnings.warn(
+    'werkzeug.contrib.atom is deprecated as of version 0.15 and will'
+    ' be removed in version 1.0.',
```
Comment (RonnyPfannschmidt) [inline werkzeug/contrib/atom.py]: is there a known alternative we could suggest

Comment (lepture) [reply werkzeug/contrib/atom.py]: How about Jinja templating

File: werkzeug/contrib/limiter.py (hunk 1 above)
Comment (lepture) [reply werkzeug/contrib/limiter.py]: If it started from 0.9. I think it is ok to remove it now.

**Comment by davidism** [comment]
I know limiter already said it was deprecated, but since it didn't have a warning maybe it would be good to add that and keep its deprecation/removal synced with the rest.

**Comment by davidism** [comment]
Turns out there's a section in the docs calling out some of the deprecations too: http://werkzeug.pocoo.org/docs/0.14/transition/#stop-using-deprecated-things

File: werkzeug/contrib/jsrouting.py (hunk 2 above)
Comment (davidism) [reply werkzeug/contrib/jsrouting.py]: From http://werkzeug.pocoo.org/docs/0.14/transition/#stop-using-deprecated-things, looks like it's already been called out as deprecated:

> stop using URL generation for JavaScript, it does not scale well with many public routing.

File: werkzeug/contrib/atom.py (hunk 4 above)
Comment (RonnyPfannschmidt) [reply werkzeug/contrib/atom.py]: pure text template engines are very horrific for making xml output

File: werkzeug/contrib/jsrouting.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply werkzeug/contrib/jsrouting.py]: thanks for that note

File: werkzeug/contrib/atom.py (hunk 4 above)
Comment (davidism) [reply werkzeug/contrib/atom.py]: Quick search turned up [python-feedgen](https://github.com/lkiesow/python-feedgen). I don't think we need to recommend things in the warnings unless they're ours.

File: werkzeug/contrib/jsrouting.py [hunk 5]
```diff
@@ -18,9 +18,18 @@
...
+    ' be removed in version 1.0.',
+    WerkzeugContribDeprecationWarning,
+    stacklevel=3
```
[the remark this answers is not in the document]
Comment (lepture) [reply werkzeug/contrib/jsrouting.py]: I want to show this warning at import for cache too.

**Comment by davidism** [comment]
`jsrouting` and `testtools` have been issuing deprecation warnings for years, I'm just going to completely remove them like `limiter`.

**Comment by davidism** [comment]
Also switching to just `DeprecationWarning` with `warnings.filterwarnings("default", category=DeprecationWarning, module="werkzeug")`. There's other things we want to deprecate in Werkzeug that aren't in contrib, and IDEs like PyCharm recognize `DeprecationWarning`.

**Pull request merged**